log/loggerInterface.go

```diff
@@ -35,6 +35,9 @@ type LogEntry struct {
 	// Data set by the user.
 	Data frozen.Map
 
+	// Description of the source code reference of the calling function
+	Caller string
```

Comment:
Any reason to choose a string? Just a suggestion but I think `Caller` should be a struct with `file` and `lineNumber` as its fields. It opens up more freedom on formatting.

Reply:
It's already formatted by the `CodeReferenceFormatter` by the time it's added to the `LogEntry`. By this stage it may no longer be a file and line number, it may be a url:
`https://github.com/example/blob/master/example.go#L42`

Reply:
is it possible to do all formatting at the end? `LogEntry` shouldn't hold any formatted data. It maybe important for user to divide the file and line number. For example, if they want to create a custom json format and they want the following format:
```js
{
    "file": "file.go",
    "line": 120,
    ...
}
```

Reply:
I agree, however the formatters are separate from the logger. If we wanted to take this approach we could either set this property against the formatters themselves or pass them through to the `Format` method. Thoughts?

Reply:
Thinking further. If the functionality is pushed to the formatters and we're only including a basic `CodeReferenceFormatter` then there's very little reason for the interface to exist at all. I'll get rid of it.